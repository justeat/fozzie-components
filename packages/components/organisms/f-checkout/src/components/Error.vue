<template>
    <card
        is-rounded
        has-outline
        is-page-content-wrapper
        card-heading-position="center"
        data-test-id="checkout-error-component"
        :class="$style['c-checkout-error']"
    >
        <img
            :src="sadBagSvg"
            alt="">

        <h1
            :class="$style['c-checkout-error-header']">
            {{ $t('errorMessages.pageLoad.heading') }}
        </h1>

        <p
            :class="$style['c-checkout-error-description']">
            {{ $t('errorMessages.pageLoad.description') }}
        </p>
    </card>
</template>

<script>
import Card from '@justeat/f-card';
import '@justeat/f-card/dist/f-card.css';

export default {
    components: { Card },
    data () {
        return {
            sadBagSvg: require('../assets/jet-sad-bag.svg') //eslint-disable-line
        };
    }
};
</script>

<style lang="scss" module>
div.c-checkout-error {
    display: flex;
    flex-direction: column;
    align-items: center;
    text-align: center;

    padding-top: spacing(x8);
    padding-left: spacing();
    padding-right: spacing();

    @include media('<=narrow') {
        border: none;
        padding-top: spacing(x2);
        padding-bottom: spacing(x2);
    }

    .c-checkout-error-header {
        @include font-size(heading-s);
        margin-top: spacing(x8);
        margin-bottom: 0;
    }

    .c-checkout-error-description {
        @include font-size(body-l);
        margin-top: spacing();
    }
}
</style>
