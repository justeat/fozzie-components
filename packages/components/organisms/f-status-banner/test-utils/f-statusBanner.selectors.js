export const COMPONENT = '[data-test-id="main-banner-container"]';
export const ADDRESS_BOX = '[data-test-id="address-box-input"]';
export const BUTTON = '[data-test-id="find-restaurants-button"]';
