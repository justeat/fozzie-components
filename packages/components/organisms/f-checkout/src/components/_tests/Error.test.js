import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import Error from '../Error.vue';

const localVue = createLocalVue();

localVue.use(Vuex);

describe('Error', () => {
    let wrapper;

    beforeEach(() => {
        // Arrange & Act
        wrapper = shallowMount(Error, {
            localVue,
            propsData: {
                header: 'Test header',
                description: 'Test description'
            }
        });
    });


    it('should be defined', () => {
        // Assert
        expect(wrapper.exists()).toBe(true);
    });

    it('should show the heading', () => {
        // Act
        const heading = wrapper.find('[data-test-id="checkout-error-page-heading"]');

        // Assert
        expect(heading.text()).toMatchSnapshot();
    });

    it('should show the description', () => {
        // Act
        const description = wrapper.find('[data-test-id="checkout-error-page-description"]');

        // Assert
        expect(description.text()).toMatchSnapshot();
    });
});
