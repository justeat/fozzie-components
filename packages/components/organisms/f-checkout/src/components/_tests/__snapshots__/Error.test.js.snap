// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should show the description 1`] = `"Test description"`;

exports[`Error should show the heading 1`] = `"Test header"`;
